diff --git a/toolchain/gcc/common.mk b/toolchain/gcc/common.mk
index 0ccf55b..2ad1a7a 100644
--- a/toolchain/gcc/common.mk
+++ b/toolchain/gcc/common.mk
@@ -100,7 +100,7 @@ GCC_CONFIGURE:= \
 		--target=$(REAL_GNU_TARGET_NAME) \
 		--with-gnu-ld \
 		--enable-target-optspace \
-		--disable-libgomp \
+		--enable-libgomp \
 		--disable-libmudflap \
 		--disable-multilib \
 		--disable-libmpx \
diff --git a/toolchain/musl/common.mk b/toolchain/musl/common.mk
index 35bdfd1..22ed296 100644
--- a/toolchain/musl/common.mk
+++ b/toolchain/musl/common.mk
@@ -27,6 +27,7 @@ include $(INCLUDE_DIR)/host-build.mk
 include $(INCLUDE_DIR)/hardening.mk
 
 TARGET_CFLAGS:= $(filter-out -O%,$(TARGET_CFLAGS))
+TARGET_CFLAGS:= $(filter-out -fopenmp,$(TARGET_CFLAGS))
 TARGET_CFLAGS+= $(if $(CONFIG_MUSL_DISABLE_CRYPT_SIZE_HACK),,-DCRYPT_SIZE_HACK)
 
 MUSL_CONFIGURE:= \
